flytekit-testing/src/main/java/org/flyte/flytekit/testing/SdkTestingExecutor.java

```diff
@@ -289,6 +290,38 @@ public <InputT, OutputT> SdkTestingExecutor withTask(
     return toBuilder().putFixedTask(task.getName(), fixedTask.withRunFn(runFn)).build();
   }
 
+  public <InputT, OutputT> SdkTestingExecutor withWorkflowOutput(
+      SdkWorkflow workflow,
+      SdkType<InputT> inputType,
+      InputT input,
+      SdkType<OutputT> outputType,
+      OutputT output) {
+    return withWorkflowOutput(
+        workflow, inputType.toLiteralMap(input), outputType.toLiteralMap(output));
+  }
+
+  public SdkTestingExecutor withWorkflowOutput(
+      SdkWorkflow workflow, Map<String, Literal> input, Map<String, Literal> output) {
+    // TODO type checking
+    TypedInterface intf = workflowTemplateMap().get(workflow.getName()).interface_();
+
+    // fixed tasks
+    SdkType<Map<String, Literal>> inputType = TestingSdkType.of(intf.inputs());
+    SdkType<Map<String, Literal>> outputType = TestingSdkType.of(intf.outputs());
+    TestingRunnableTask<Map<String, Literal>, Map<String, Literal>> fixedTask =
+        getFixedTaskOrDefault(workflow.getName(), inputType, outputType);
+
+    // replace workflow
+    SdkWorkflowBuilder builder = new SdkWorkflowBuilder();
+    SdkWorkflow w = new TestingWorkflow(workflow, inputType, outputType);
```

Comment:
```suggestion
    SdkWorkflow testingWorkflow = new TestingWorkflow(workflow, inputType, outputType);
```
I think it is clearer to differentiate it from `workflow` variable.